java/CustomMetrics/README.md

```diff
@@ -3,18 +3,34 @@
 * Flink version: 1.20
 * Flink API: DataStream API
 * Language: Java (11)
+* Flink connectors: Kinesis Sink
 
 This example demonstrates how to create your own metrics to track application-specific data, such as processing events or accessing external resources and publish it to Cloudwatch.
 
+The applications generates data internally and writes to a Kinesis Stream.
+
 ### Runtime configuration
 
-The application reads the runtime configuration from the Runtime Properties, when running on Amazon Managed Service for Apache Flink.
+When running on Amazon Managed Service for Apache Flink the runtime configuration is read from *Runtime Properties*.
+
+When running locally, the configuration is read from the [`resources/flink-application-properties-dev.json`](resources/flink-application-properties-dev.json) file located in the resources folder.
+
+Runtime parameters:
+
+| Group ID        | Key           | Description               | 
+|-----------------|---------------|---------------------------|
+| `OutputStream0` | `stream.name` | Name of the output stream |
+| `OutputStream0`  | `aws.region`  | (optional) Region of the output stream. If not specified, it will use the application region or the default region of the AWS profile, when running locally. |
+
+All parameters are case-sensitive.
+
+This simple example assumes the Kinesis Stream is in the same region as the application, or in the default region for the authentication profile, when running locally.
 
-Runtime Properties are expected in the Group ID `OutputStream0`. They are all case-sensitive:
-* `stream.name` - Kinesis Data Stream to be used for sink.
-* `aws.region` - AWS Region containing test resources.
 
 ### Running locally in IntelliJ
-Update `PropertyMap` in [configuration file](src/main/resources/flink-application-properties-dev.json).
 
-To start the Flink job in IntelliJ edit the Run/Debug configuration enabling *'Add dependencies with "provided" scope to the classpath'*.
+To run the application locally, in IntelliJ:
+
+1. Update `PropertyMap` in [configuration file](src/main/resources/flink-application-properties-dev.json).
+2. Edit the Run/Debug configuration enabling *'Add dependencies with "provided" scope to the classpath'*.
+3. Use the [AWS Toolkit](https://aws.amazon.com/intellij/) plugin to run the application with an AWS profile with access to the destination Kinesis Stream.
```

Comment:
do we also have a centralized place we can reference for guide to setting up locally?

Reply:
That's a good point.
Let me create a centralized .md file with instructions to run Java examples locally

Reply:
@jeremyber-aws I centralized the "How to run in IntelliJ" in a separate md file
Please have a look